react/components/SearchBar/components/SearchBar.js

```diff
@@ -97,6 +114,36 @@ const SearchBar = ({
     [navigate, attemptPageTypeSearch, customSearchPageUrl]
   )
 
+  const validateInput = () => {
+    if (minSearchTermLength && inputValue.length < minSearchTermLength) {
+      return intl.formatMessage({
+        id: 'store/search.searchTermTooShort',
+      })
+    }
+
+    return undefined
+  }
+
+  const showInputErrorMessageTemporarily = inputErrorMessage => {
+    setInputErrorMessage(inputErrorMessage)
+
+    if (inputErrorMessageTimeRef.current) {
+      clearTimeout(inputErrorMessageTimeRef.current)
+    }
+
+    inputErrorMessageTimeRef.current = setTimeout(() => {
+      setInputErrorMessage()
+    }, INPUT_ERROR_MESSAGE_DELAY)
+  }
+
+  const hideInputErrorMessage = () => {
+    if (inputErrorMessageTimeRef.current) {
+      clearTimeout(inputErrorMessageTimeRef.current)
+    }
+
+    setInputErrorMessage()
+  }
```

Comment:
I think the error message should be static and only be removed after user typing.